### PR TITLE
Fix forward AD copy_ into same-sized tensor without fw grad

### PR DESCRIPTION
**Pull request by soulitzer**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #79767
* #79664
* __->__ #79653

### CODE REVIEW

File: torch/csrc/autograd/VariableTypeManual.cpp [hunk 1]
```diff
@@ -214,7 +214,7 @@ Tensor& copy_(
       if (!self.is_same_size(src_fw_grad)) {
         new_fw_grad = src_fw_grad.broadcast_to(self.sizes());
       } else {
-        new_fw_grad = src_fw_grad;
+        new_fw_grad = src_fw_grad.clone();
```
Comment (zou3519) [inline torch/csrc/autograd/VariableTypeManual.cpp:217]: This is the same situation as https://github.com/pytorch/pytorch/pull/79664#discussion_r899198907 right? If so I'll defer to what the outcome of that discussion is

**Review by albanD — APPROVED** [review]
Following the other discussion, that sounds good!

**Pull request closed without merging**